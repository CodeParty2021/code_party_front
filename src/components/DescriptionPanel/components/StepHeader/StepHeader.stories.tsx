import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { StepHeader } from "./StepHeader";

export default {
  title: "components/DescriptionPanel/StepHeader",
  component: StepHeader,
} as ComponentMeta<typeof StepHeader>;

const Template: ComponentStory<typeof StepHeader & { children: any }> = (
  args
) => <StepHeader {...args} />;

export const Default = Template.bind({});
Default.args = {
  step: 1,
  maxStep: 3,
  title: "チュートリアル",
};
