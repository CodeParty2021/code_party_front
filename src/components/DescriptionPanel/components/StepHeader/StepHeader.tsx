import { RoundedButton } from "components/RoundedButton/RoundedButton";
import React from "react";
import {
  ButtonBox,
  Step,
  StepHeaderStyle,
  StepHeaderStyleProps,
  TextBox,
  Title,
} from "./StepHeaderStyle";

type Props = StepHeaderStyleProps & {
  step: number;
  maxStep: number;
  title: string;
};
export const StepHeader: React.FC<Props> = ({
  step,
  maxStep,
  title,
  ...styleProps
}) => {
  return (
    <StepHeaderStyle {...styleProps}>
      <TextBox>
        <Step>Step {step}</Step>
        <Title>{title}</Title>
      </TextBox>
      <ButtonBox>
        <RoundedButton
          icon="left"
          onClick={() => {}}
          value="前へ"
          disabled={step === 1}
        />
        <RoundedButton
          icon="right"
          onClick={() => {}}
          value="次へ"
          disabled={step === maxStep}
        />
      </ButtonBox>
    </StepHeaderStyle>
  );
};
