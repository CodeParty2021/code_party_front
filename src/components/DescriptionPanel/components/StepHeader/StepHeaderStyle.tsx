import styled from "styled-components";
import { descriptionPanel } from "styles/colors";
import { FlexGap } from "styles/FlexGap/FlexGap";
export type StepHeaderStyleProps = {};

export const StepHeaderStyle = styled.div`
  display: flex;
  ${FlexGap({ gap: "24px", direction: "row" })}
  justify-content: space-between;
  width: 100%;
`;
export const ButtonBox = styled.div`
  display: flex;
  ${FlexGap({ gap: "8px", direction: "row" })}
`;
export const Step = styled.div`
  font-family: "851Gkktt";
  font-style: normal;
  font-weight: 400;
  font-size: 40px;
  line-height: 100%;
  /* identical to box height, or 40px */

  color: ${descriptionPanel.stepHeader.stepColor};
`;

export const TextBox = styled.div`
  display: flex;
  ${FlexGap({ gap: "24px", direction: "row" })}
  flex-direction: row;
  align-items: center;
`;
export const Title = styled.div`
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 26px;
  /* identical to box height */

  font-feature-settings: "palt" on;

  color: ${descriptionPanel.stepHeader.titleColor};

  transform: matrix(1, 0, -0.08, 1, 0, 0);
`;
