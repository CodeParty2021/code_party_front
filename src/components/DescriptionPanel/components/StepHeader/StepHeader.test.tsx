import React from "react";
import { StepHeader } from "./StepHeader";
import "jest-styled-components";
import { render } from "enzyme";

describe("<StepHeader />", () => {
  it("snapshot test", () => {
    const wrapper = render(
      <StepHeader {...{ step: 0, maxStep: 3, title: "チュートリアル" }} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
