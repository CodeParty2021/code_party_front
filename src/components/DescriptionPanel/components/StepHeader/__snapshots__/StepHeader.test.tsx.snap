// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepHeader /> snapshot test 1`] = `
.c6 {
  display: inline-block;
  width: 16px;
  height: 16px;
  fill: #2F364D;
}

.c6 > span,
.c6 > div,
.c6 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  height: 40px;
  background: #F1F3F8;
  border: 1px solid #E1E3EB;
  border-radius: 40px;
  color: #2F364D;
  font-size: 16px;
  line-height: 160%;
  padding: 4px 16px 4px 12px;
}

.c5:hover {
  border-color: #6074FF;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  height: 40px;
  background: #F1F3F8;
  border: 1px solid #E1E3EB;
  border-radius: 40px;
  color: #2F364D;
  font-size: 16px;
  line-height: 160%;
  padding: 4px 12px 4px 16px;
}

.c7:hover {
  border-color: #6074FF;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 24px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

.c2 {
  font-family: "851Gkktt";
  font-style: normal;
  font-weight: 400;
  font-size: 40px;
  line-height: 100%;
  color: #6074ff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 24px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 26px;
  font-feature-settings: "palt" on;
  color: #4d5366;
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
}

@supports not (gap:8px) {
  .c5 > *:not(:first-child) {
    margin-left: 8px;
  }
}

@supports not (gap:8px) {
  .c7 > *:not(:first-child) {
    margin-left: 8px;
  }
}

@supports not (gap:24px) {
  .c0 > *:not(:first-child) {
    margin-left: 24px;
  }
}

@supports not (gap:8px) {
  .c4 > *:not(:first-child) {
    margin-left: 8px;
  }
}

@supports not (gap:24px) {
  .c1 > *:not(:first-child) {
    margin-left: 24px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Step 0
    </div>
    <div
      class="c3"
    >
      チュートリアル
    </div>
  </div>
  <div
    class="c4"
  >
    <button
      class="c5"
    >
      <span
        class="c6"
        display="inline-block"
        fill="#2F364D"
      />
      <span>
        前へ
      </span>
    </button>
    <button
      class="c7"
    >
      <span>
        次へ
      </span>
      <span
        class="c6"
        display="inline-block"
        fill="#2F364D"
      />
    </button>
  </div>
</div>
`;
